src/System.Data.SqlClient/src/System/Data/SqlClient/SqlStatistics.cs

```diff
@@ -218,6 +221,100 @@ internal void UpdateStatistics()
                 _connectionTime = long.MaxValue;
             }
         }
+
+        // We subclass Dictionary to provide our own implementation of CopyTo, Keys.CopyTo, and
+        // Values.CopyTo to match the behavior of Hashtable, which is used in the full framework.
+        // Ideally this would derive from Dictionary<string, long>, but that would break compatibility
+        // with the full framework, which allows adding keys/values of any type.
+        private sealed class StatisticsDictionary : Dictionary<object, object>, IDictionary
```

Comment:
I updated the comment with the behavior differences.

BTW, I'm personally not a fan of having to include that dictionary subclass, I'm just being very pedantic about matching the previous behavior. In an ideal world, we'd just change this to return `Dictionary<string, long>` and live with the minor behavior differences between .NET Framework and .NET Core. My guess is the impact would be minimal. Would this be something we could consider doing?